`Web docs` Adding description list patterns doc (#3029)
hashicorp/design-system: website/docs/patterns/description-list-patterns/partials/guidelines/guidelines.md

!!!

### Alignment

We recommend that content is left aligned, even if the values are related to other content on the page. This maintains the natural reading flow of the content.

![A description list with three columns contained in a card sitting above a table. Each list items is aligned to the left while the values in the table in the last columnn are right aligned.](/assets/patterns/description-list-patterns/description-list-alignment-left.png)
 
## Orientation

A description list can be oriented either vertically or horizontally.

### Vertical

Use vertical orientation for designs similar to metacards or summary card experiences. With this orientation, the value in the description list can be styled either using the standard or the callout style.

#### Spacing between the heading and value

Use an 8px gap in between the heading and value. Components can be used in place of the value, but only one component per description list item is allowed.

![Two rows of five columns with description list items. First row using the standard style while the second row using the callout style.](/assets/patterns/description-list-patterns/description-list-vertical-spacing-heading-value.png)

> In rare instances where the value slot requires additional rows for context, the gap is 4px. The content within the value slot can have multiple rows. Consider the visual hierarchy of the overall grouping during implementation.

Reviewer: [suggestion] this more consistent with the other language that has been pivoted to say "use"

```suggestion
In rare instances where the value slot requires additional rows for context, use a 4px gap. The content within the value slot can have multiple rows. Consider the visual hierarchy of the overall grouping during implementation.
```

!!!

### Horizontal

Horizontal orientation should only be used with the standard styling of headings and values. Using two sizes in the horizontal orientation throws off the visual balance of the content in a description list.

#### Spacing

Use 16px gaps between the heading and value and each row pairing. Components can be used in place of the value, but only one component should be used per description list item.

![Five rows of information showing a vertical gap between the heading and value of 16px and between each row as 16px.](/assets/patterns/description-list-patterns/description-list-horizontal-standard-grid-gaps.png)

In rare instances where the value slot requires additional rows for context, use a 4px gap. The content within the value slot can have multiple rows. Consider the visual hierarchy of the overall grouping during implementation.

![The gap between the heading and value is 16px. The value has two rows and a 4px gap between each row.](/assets/patterns/description-list-patterns/description-list-horizontal-value-multiple-rows.png)

##### Exceptions

> There may be occasions when the standard gap between items in a description list needs to be adjusted. For example, a description list used as a legend requires a 4px row gap instead of the standard 16px, as the values are closely related.

Reviewer: Maybe I missed something here, but should this be 8px?

```suggestion
There may be occasions when the standard gap between items in a description list needs to be adjusted. For example, a description list used as a legend requires a 4px row gap instead of the standard 8px, as the values are closely related.
```

![Highlighting list items in a horizontal orientation stacked vertically on top of each other with a 4px gap between each other.](/assets/patterns/description-list-patterns/description-list-logical-gap.png)

> In horizontal orientations, use your best judgment to adjust the gaps between list items for optimal UI representation.

Reviewer: [suggestion, non-blocking] "hierarchy" seems a bit more correct here, but not a blocker

```suggestion
In horizontal orientations, use your best judgment to adjust the gaps between list items for optimal UI hierarchy.
```

> #### Column width

Reviewer: Here's where I think things a bit confusing (we chatted about this briefly today), but the `Description List` is the pattern consisting of a list of `Description List / Items` (thinking about organization here), so is this actually the **Item** that is organized horizontally, and the **List** that is organized vertically? Might be good to be a bit more clear about that (may impact some other parts of the documentation).

When the description list is oriented horizontally, content length will vary. If the heading and value’s content length is fairly similar, then the width of both the heading and value should be set to fill (50%) of the available space. This often happens when a description list is placed within a card.

![A card titled "Cluster details" with a description list below it. Each list item has its heading and value's width at 50% of the cards space.](/assets/patterns/description-list-patterns/description-list-horizontal-fill-example.png)

> If the values’ content is much longer than the heading, set the heading’s width to fixed (equal to the longest content length within the column), while the value is set to fill (100% of the remaining space). This often happens when a description list is placed alone on the page as the primary content.

Reviewer: I'm not sure of the grammatical side here, but above you use "value's" with regard to the plurality of heading and value, but here you use "values'". I'm not sure which one is right, but it seemed weird to me.

Author: @andgen404 also pointed this out in the draft and I said I'm attempting to mention the overall grouping of values. Because its not just about one value being longer, it's all of them.

> ## Typography

Author: I feel that may be too much since we are already explaining it below within the value section?

> ## Typography

Author: @LilithJames-HDS how does this feel now (look below)?

This style is used to emphasize values over headings. It is often used for numerical content, like resources or financial data. Use `Display 300 / 400 Bold` (depending on content hierarchy) with `Foreground / Strong`. This should only be used for vertical orientation list items.

![](/assets/patterns/description-list-patterns/description-list-value-callout.png)

!!! Dont

Mix and match visual styles. Ensure visual consistency is maintained by choosing one style in a given description list.

![A card with five columns of headings and values. The last column using the callout style while the rest using the standard style.](/assets/patterns/description-list-patterns/description-list-dont-mix-match-styles.png)

!!!

### Alignment

We recommend that content is left aligned, even if the values are related to other content on the page. This maintains the natural reading flow of the content.

![A description list with three columns contained in a card sitting above a table. Each list items is aligned to the left while the values in the table in the last columnn are right aligned.](/assets/patterns/description-list-patterns/description-list-alignment-left.png)
 
## Orientation

> A list item can be oriented either vertically or horizontally. Use one orientation per description list.

Author: Do we feel it is necessary to add that here? It is explained that in horizontal only one style is allowed, while vertical allows for both the callout and standard styles.

When using the callout value style, use a grid with 24px vertical gaps and 48px horizontal gaps. Set each list item’s width to fill for even visual spacing.

![Two rows with three columns. The gaps vertically show 24px and horizontally shows 48px.](/assets/patterns/description-list-patterns/description-list-vertical-callout-grid-gaps.png)

#### Wrapping

If content is too condensed, wrap content to a new row.

!!! Do

Wrap content to ensure legibility of each list item.

![Two rows with five columns. The first row has each column filled, while the second row has one. Information fits comfortably in each column](/assets/patterns/description-list-patterns/description-list-vertical-do-grid.png)

!!!

!!! Dont

Force list items into a single row, instead consider arranging items into uniform rows. This practice promotes visual consistency and enhances readability.

![One row with seven columns. The information in each column wraps inconsistently and has varied heights](/assets/patterns/description-list-patterns/description-list-vertical-dont-grid.png)

!!!

> ### Horizontal

Reviewer 2: I think the page structure should be 

I. Description List styles (standard, callout)
II. Typography
III. Orientation

> ### Horizontal

Author: Ok, this will require some rework on the last day this is due. Bare with me a bit.

> ### Horizontal

Reviewer 3: @LilithJames-HDS I'm curious to understand how the content in the proposed "Description List styles" section differs from what is in the "Typography" section. Can you clarify?

> ### Horizontal

Reviewer 3: > Also keeping in mind this doc is due today and would rather not delay its completion for a restructure that may not be necessary.

@majedelass We shouldn't ship content that lacks alignment or is confusing to others. Let's lean into our key principle of [Quality by default](https://helios.hashicorp.design/about/principles#:~:text=03-,Quality%20by%20default,-We%20recognize%20that) and work towards clarity.

> ### Horizontal

Reviewer 2: It just felt like hopping right into the deep end rather than outlining the options then diving into the type and layout of those options

> ### Horizontal

Author: I see what you're saying. Let me make an adjustment and see how that feels.

> ### Horizontal

Reviewer 3: @LilithJames-HDS I think so. While I was reading, I also [wondered](https://github.com/hashicorp/design-system/pull/3029#pullrequestreview-3030165150) if it would make sense to have a generic section above typography that would clarify the overall pattern options before diving into the nitty gritty.

> ### Horizontal

Reviewer 2: Ah ok great, yes we are on the same page then 💡

> ### Horizontal

Author: How does this read now [(look below this comment)](https://github.com/hashicorp/design-system/pull/3029/files#r2216051907)

> ### Horizontal

Author: I have an idea, one sec.

> Horizontal orientation should only be used with the standard styling of headings and values. Using two sizes in the horizontal orientation throws off the visual balance of the content in a description list.

Reviewer 3: ```suggestion
Horizontal orientation should only be used with the standard value styling. Using two sizes in the horizontal orientation throws off the visual balance of the content in a description list.
```

#### Spacing

> Use 16px gaps between the heading and value and each row pairing. Components can be used in place of the value, but only one component should be used per list item.

Reviewer 3: ```suggestion
Use 16px gaps between the heading and value and each row. 

While text is most common, other elements (like `Badge` , `CopySnippet`, links, etc) may also be used for the value. We recommend one element per value to keep things simple and related to the heading.
```

![Five rows of information showing a vertical gap between the heading and value of 16px and between each row as 16px.](/assets/patterns/description-list-patterns/description-list-horizontal-standard-grid-gaps.png)

> In rare instances where the value slot requires additional rows for context, use a 4px gap. The content within the value slot can have multiple rows. Consider the visual hierarchy of the overall description list during implementation.

Reviewer 3: ```suggestion
In instances where the value may require additional rows for context, use a 4px gap. When selecting a text style and color, consider the visual hierarchy of the overall description list.
```

![The gap between the heading and value is 16px. The value has two rows and a 4px gap between each row.](/assets/patterns/description-list-patterns/description-list-horizontal-value-multiple-rows.png)

##### Exceptions

> There may be occasions when the standard gap between items in a description list needs to be adjusted. For example, a description list used as a legend requires a 4px row gap instead of the standard 16px, as the values are closely related.

Reviewer 3: ```suggestion
On occasion, the recommended gap size may not be suitable for the layout. For example, when a description list is used as a legend. In this case, a 4px gap could be appropriate as the content is more closely related than in a standard description list.
```

![Highlighting list items in a horizontal orientation stacked vertically on top of each other with a 4px gap between each other.](/assets/patterns/description-list-patterns/description-list-logical-gap.png)

In horizontal orientations, use your best judgment to adjust the gaps between list items for optimal UI hierarchy.

> #### Width definition

Reviewer 3: ```suggestion
#### Column width
```

> When the list item is oriented horizontally, content length will vary. If the heading and value’s content length is fairly similar, then the width of both the heading and value should be set to fill (50%) of the available space. This often happens when the description list is placed within a card.

Reviewer 3: ```suggestion
When the list item is oriented horizontally and the content length of the heading and value is fairly similar, we recommend using equal column widths (50%) to fill the available space.
```

![A card titled "Cluster details" with a description list below it. Each list item has its heading and value's width at 50% of the cards space.](/assets/patterns/description-list-patterns/description-list-horizontal-fill-example.png)

> If the value's content is much longer than the heading, set the heading’s width to fixed (equal to the longest content length within the column), while the value is set to fill (100% of the remaining space). This often happens when a description list is placed alone on the page as the primary content.

Reviewer 3: ```suggestion
If the content length of the value is considerably longer than the heading, we recommend that the heading column width be equal to the longest heading content, while the value uses the rest of the available space.
```

## Typography

> List items have two visual style options: standard or callout. Both share typography styling for the _title_ but the typography for the _value_ is adjusted.

Reviewer 3: I know we don't typically have a heading followed by a heading, but I can't come up with anything helpful that is generic to both title and value. 
```suggestion
```

> List items can be visually styled one of two ways: standard or callout. In either of the options, the title's typography is maintained, while the value is adjusted.

Reviewer 2: ```suggestion
List items have two visual style options: standard or callout. Both share typography styling for the _title_ but the typography for the _value_ is adjusted.
```
This maybe is a little clearer/simpler?

Author: Much better, thanks!

## Typography

### Title

The title should always use the same styling of  `Body 200 Semibold` with `Foreground / Primary`, regardless of the orientation or value style.

![](/assets/patterns/description-list-patterns/description-list-heading.png)

### Value

Values have two visual style options: standard or callout. 

#### Standard

Use `Body 200 Regular` with `Foreground / Strong` for values when the title and value are of equal importance.

![](/assets/patterns/description-list-patterns/description-list-value-standard.png)

#### Callout

Use `Display 300 Bold` or `Display 400 Bold` (depending on existing content hierarchy) with `Foreground / Strong` when the value should be emphasized over the title. The callout style is often used for numerical content, like resources or financial data. 

The callout style should only be used in the vertical orientation.

![](/assets/patterns/description-list-patterns/description-list-value-callout.png)

!!! Dont

> Mix and match visual styles. Ensure visual consistency is maintained by choosing one style in a given description list.

Reviewer 4: If a user doesn't read this together with the red "Don't" tag in the callout, then this reads as telling them they _should_ mix and match styles. To avoid ambiguity with this type of sentence, we've tended to include the word "Don't" within the sentence to ensure it's read together.

Example: https://helios.hashicorp.design/patterns/form-patterns#multi-column

Author: Will adjust, thanks for the catch!

![Two rows of one column with description list items. Each description list item has a title and a value with two rows of information. First row uses the standard style while the second row uses the callout style.](/assets/patterns/description-list-patterns/description-list-vertical-value-multiple-lines.png)

#### Spacing between list items

When using the standard value style, use a grid with 24px gaps both vertically and horizontally. When placed inside a visible container, we recommend spacing the list items evenly. 

![Two rows with five columns. The first row has each column filled, while the second row has one. The gaps vertically and horizontally shows 24px gaps.](/assets/patterns/description-list-patterns/description-list-vertical-grid-gaps.png)

When using the callout value style, use a grid with 24px vertical gaps and 48px horizontal gaps. When placed inside a visible container, we recommend spacing the list items evenly.

![Two rows with three columns. The gaps vertically show 24px and horizontally shows 48px.](/assets/patterns/description-list-patterns/description-list-vertical-callout-grid-gaps.png)

#### Wrapping

> If content is too condensed, wrap content to a new row.

Reviewer: [nit] this prevents using content twice in the same sentence.

```suggestion
If content is too condensed, wrap list items to a new row.
```
